### ISSUE TITLE
Loss goes to NaN while using fp16 when working with arcface_torch 

### ISSUE THREAD

**Issue by stavyadatta** [deepinsight/insightface#2200]
Hello
I am trying to train an arcface model on my custom dataset with 240226 ids. Here is my config file 

`from easydict import EasyDict as edict

# make training faster
# our RAM is 256G
# mount -t tmpfs -o size=140G  tmpfs /train_tmp

config = edict()
config.margin_list = (1.0, 0.0, 0.4)
config.network = "r50"
config.resume = False
config.output = None
config.embedding_size = 512
config.sample_rate = 1.0
config.fp16 = False
config.momentum = 0.9
config.weight_decay = 1e-4
config.batch_size = 224
config.lr = 0.1
config.verbose = 2000
config.dali = False

config.rec = "/workspace/awi_facial_recognition/test_set_2/"
#config.use_pretrained = True
#config.pretrained_model_path = "/workspace/16backbone.pth"
config.num_classes = 1298
config.num_image = 38048
config.num_epoch = 20
config.warmup_epoch = 0
config.val_targets = ['lfw', 'cfp_fp', 'agedb_30', 'calfw.bin']`

After training for a couple of hours, I receive an error during validation. The error is regarding Nan values in the embedding. My loss also goes to nan before this error occurs. Can anyone please help on this? 

This error occurs before a whole epoch is completed. Please help me if possible.

**Comment by tessarwork** [comment]
hello i got same problem, did you solve it?
